Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Managers/Payment Method Managers/PrimerHeadlessMainComponentWrapper.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  PrimerHeadlessMainComponentWrapper.swift
+//  PrimerSDK
+//
+//  Created by Alexandra Lovin on 22.11.2023.
+//
+
+import Foundation
+@objc public protocol BanksComponentErrorable: AnyObject {
+    @objc func didReceiveError(_ error: NSError)
+}
+
+@objc public protocol BanksComponentSteppable: AnyObject {
+    @objc func didStartLoading()
+    @objc func didReceiveBanks(_ banks: [IssuingBank])
+}
+
+@objc public enum ValidStatus: Int {
+    case validating
+    case valid
+    case invalid
+    case error
+}
+
+@objc public protocol BanksComponentValidatable: AnyObject {
+    @objc func didReceiveValidationStatus(_ status: ValidStatus)
+}
+
+@objc public protocol PrimerHeadlessMainComponentWrapperDelegate: AnyObject, PrimerHeadlessSubmitable, PrimerHeadlessStartable {
+    @objc var stepDelegate: BanksComponentSteppable? { get set }
+    @objc var errorDelegate: BanksComponentErrorable? { get set }
+    @objc var validationDelegate: BanksComponentValidatable? { get set }
+    @objc func selectBankById(_ bankdId: String)
+    @objc func filterBankByName(_ text: String)
+}
+
+@objc public final class PrimerHeadlessMainComponentWrapper: NSObject, PrimerHeadlessMainComponentWrapperDelegate {
+    var banksComponent: (any BanksComponent)?
```

Comment:
Is this correct? It seems like there is a tight coupling here between `PrimerHeadlessMainComponentWrapper` and `BanksComponent`.

Reply:
Indeed, there a coupling between the wrapper and the banks component in this case. But this is only for the ObjC compatibility wrapper (I wanted all of the interaction with the bank component to happen with the help of the PrimerHeadlessMainComponentWrapper)
This way, the ObjC client can use the PrimerHeadlessMainComponentWrapper (which internally forwards all requests to the BanksComponent). 
But with addition of new components / per different payment methods, the ObjC compatibility is going to be a bit challenging. Maybe better to use a separate specialized methods, eg: 
@objc public func provideBanksComponentWrapper(paymentMethodType: String) -> PrimerHeadlessBanksComponentWrapper 
where PrimerHeadlessBanksComponentWrapper , still keep an private property with the BanksComponent